### ISSUE TITLE
__time not working with Javascript aggregator (0.6.171)

### ISSUE THREAD

**Issue by schapirama** [druid-io/druid#1086]
As discussed in the Google Group (https://groups.google.com/forum/#!topic/druid-development/RL79zFn56Pw) I am not able to get a valid value out of __time in my Javascript aggregator --the value that gets passed in to the JS function is null. 

Here is a sample query:

```
{
    "intervals": [
        "2015-02-03T18:00:00.000+00:00/2015-02-03T19:00:00.000+00:00"
    ],
    "aggregations": [
        {
            "type": "count",
            "name": "cnt"
        },
        {
            "type": "javascript",
            "fieldNames": [
                "__time"
            ],
            "name": "aggtime",
            "fnReset": "function reset() { return 0 }",
            "fnCombine": "function combine(a,b) { return a + b }",
            "fnAggregate": "function aggregate(current, time) {if(time > 1) {return(current + 1);} else {return(current);}}"
        }
    ],
    "dataSource": "operational_events",
    "granularity": "all",
    "queryType": "timeseries"
}
```
Notice that I'm comparing time > 1 ... should always be true. Alas, the result I get is

```
[ {
  "timestamp" : "2015-02-03T18:22:00.000Z",
  "result" : {
    "aggtime" : 0.0,
    "cnt" : 22
  }
} ]
```

This is against 0.6.171.

For completeness, in case I've done something wrong, here's the ingest spec (created for this test)

```
  {
    "type" : "index_realtime",
    "schema": {
      "dataSource": "operational_events_temp",
      "aggregators": [
        {
          "type": "count",
          "name": "rowCount"
        }
      ],
      "indexGranularity": "minute",
      "shardSpec": {
        "type": "none"
      }
    },
    "config": {
      "maxRowsInMemory": 500000,
      "intermediatePersistPeriod": "PT10m"
    },
    "firehose": {
      "type": "kafka-0.8",
      "consumerProps": {
        "group.id": "xxx",
        "zookeeper.connect": "xxxx",
        "zookeeper.connection.timeout.ms": "15000",
        "zookeeper.session.timeout.ms": "15000",
        "zookeeper.sync.time.ms": "5000",
        "fetch.message.max.bytes": "1048586",
        "auto.offset.reset": "largest",
        "auto.commit.enable": "true"
      },
      "feed": "operational_events",
      "parser": {
        "timestampSpec": {
          "column": "timestamp",
          "format": "millis"
        },
        "data": {
          "format": "json"
        },
        "dimensionExclusions": [
          "timestamp"
        ]
      }
    },
    "plumber": {
      "type": "realtime",
      "windowPeriod": "PT10m",
      "segmentGranularity": "hour"
    }
  }
```

and here's a segment query for the segment I am trying to query (to make sure it's a historical segment)

```
{
"queryType":"segmentMetadata",
"dataSource":"operational_events_temp",
"intervals":["2015-02-03T18:00:00.000+00:00/2015-02-03T19:00:00.000+00:00"],
"toInclude":{ "type": "list", "columns":["__time"]}
}
```

==> 
```
[ {
  "id" : "operational_events_temp_2015-02-03T00:00:00.000Z_2015-02-04T00:00:00.000Z_2015-02-03T18:22:46.586Z",
  "intervals" : [ "2015-02-03T00:00:00.000Z/2015-02-03T23:59:00.000Z" ],
  "columns" : {
    "__time" : {
      "type" : "LONG",
      "size" : 2880,
      "cardinality" : null,
      "errorMessage" : null
    }
  },
  "size" : 36971
} ]
```

Thanks.

**Comment by schapirama** [comment]
![image](https://cloud.githubusercontent.com/assets/57937/6042270/97788ae0-ac83-11e4-9081-c51a0f169acb.png)

![image](https://cloud.githubusercontent.com/assets/57937/6042332/13db92a8-ac84-11e4-97e2-2cda18c5ee7b.png)

![image](https://cloud.githubusercontent.com/assets/57937/6042340/25ced0c4-ac84-11e4-9ea5-c4499cced8fd.png)

Tiny pics... but they show that the queried segments are indeed historical (right?)

**Comment by xvrl** [comment]
Hi @schapirama, the ability to query the `__time` column is available in Druid 0.7 but has not been back-ported to 0.6. We can look into back-porting the feature, but you might be better off trying 0.7.0-rc2 if you are still experimenting. Apologies for the confusion.

**Comment by xvrl** [comment]
This has been resolved, please see
https://github.com/druid-io/druid/issues/1086 for more information.
